src/lib/gadgets/bitwise.ts

```diff
@@ -252,6 +247,10 @@ function rot(
     }
   );
 
+  // flush zero var to prevent broken gate chain (zero is used in rangeCheck64)
+  // TODO this is an abstraction leak, but not clear to me how to improve
+  toVar(0n);
```

Comment:
@querolita I think my tests already exhibit that case, because I am creating random linear combinations of variables as inputs, something like `x + 20y + 3z + 17`, and this will generate an even or odd number of generic gates when passed to a gate, depending on the length of the linear combination. Since we test many random inputs, I'm pretty sure all cases are covered

Reply:
> In OCaml we had a PR where whenever you add a gate it adds a "shadow row" that specifies what the next row should be with Some or None (for anything), per cell in the next row.

That sounds interesting! Do you know where that PR is?